src/jdk.compiler/share/classes/com/sun/tools/javac/code/Type.java

```diff
@@ -2040,6 +2088,8 @@ public CapturedType(Name name,
             this.lower = Assert.checkNonNull(lower);
             this.setUpperBound(upper);
             this.wildcard = wildcard;
+            this.universal = upper.hasTag(TYPEVAR) && ((TypeVar)upper).universal ||
```

Comment:
This probably requires tweaking - I see two issues:
* right now it detects cases where upper/lower are type variables themselves - but doesn't detect cases originating from e.g. `List<? extends Point>` - where you just have a captured variable whose bound is `Point`.
* the logic does not recurse - but maybe that's ok if `universal` is always set on creation

I think another approach could be to look at the wildcard type associated with the captured type. The wildcard type keeps track of the declared type variable it comes from, which means you can then look at the type symbol in there and decide if it's an universal variable or not. E.g.

```
captured.isUniversal() == captured.wildcard.bound.isUniversal()
```